File: src/main/java/cn/labzen/web/paging/convert/NonePageConverter.java
package cn.labzen.web.paging.convert;

import cn.labzen.web.paging.PageConverter;
import cn.labzen.web.paging.Pagination;
import cn.labzen.web.paging.internal.Paging;

public class NonePageConverter implements PageConverter<Paging> {

  @Override
  public Paging to(Paging paging) {
    return paging;
  }

  @Override
  public Pagination<?> from(Paging paging) {
    return new Pagination<>(!paging.unpaged(), paging.pageNumber(), paging.pageSize(), 0, 0, null);
  }
}

File: src/main/kotlin/cn/labzen/web/paging/converter/NonePageConverter.kt
package cn.labzen.web.paging.converter

import cn.labzen.web.paging.PageConverter
import cn.labzen.web.paging.Pagination
import cn.labzen.web.paging.internal.Paging

class NonePageConverter : PageConverter<Paging> {

  override fun to(paging: Paging): Paging = paging

  override fun from(page: Paging): Pagination<*> =
    with(page) {
      Pagination<Any>(pageNumber(), pageSize(), 0, 0)
    }
}